src/tools/log-parser-rs/src/main.rs

```diff
@@ -0,0 +1,66 @@
+#![warn(unused_crate_dependencies)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+
+//!`kata-log-parser` is a tool that combines logfiles generated by the various
+//!system components, sorts them by timestamp, and re-displays the log entries. A
+//!time delta is added to show how much time has elapsed between each log entry.
+//!
+//!The tool is also able to check the validity of all log records, can re-format the
+//!logs, and output them in a different format.
+
+mod args;
+mod log_message;
+mod log_parser_error;
+mod output_file;
+mod parse_file;
+mod process_logs;
+
+use crate::args::Cli;
+use crate::log_message::StrictLogMessage;
+use crate::log_parser_error::LogParserError;
+use crate::output_file::*;
+use crate::parse_file::*;
+use crate::process_logs::*;
+use clap::Parser;
+use log_message::AnyLogMessage;
+use log_message::LogMessage;
+use std::process::exit;
+
+fn real_real_main<'a, T: AnyLogMessage<'a>>(cli: Cli) -> Result<(), LogParserError> {
```

Comment:
Had to add another layer in order to make main generic over LogMessage and StrictLogMessage. Not sure what to name this, went with something silly *for now*.